pkg/agent/route/route_linux.go

```diff
@@ -324,86 +244,77 @@ func (c *Client) Reconcile(podCIDRs []string) error {
 	}
 	for _, entry := range entries {
 		if desiredPodCIDRs.Has(entry) {
+			klog.Infof("Reconcile has entry %s", entry)
 			continue
 		}
-		klog.V(4).Infof("Deleting orphaned ip %s from ipset", entry)
+		klog.Infof("Deleting orphaned ip %s from ipset", entry)
 		if err := ipset.DelEntry(antreaPodIPSet, entry); err != nil {
 			return err
 		}
+		_, cidr, err := net.ParseCIDR(entry)
+		if err != nil {
+			return err
+		}
+		route := &netlink.Route{Dst: cidr}
+		if err := netlink.RouteDel(route); err != nil && err != unix.ESRCH {
+			return err
+		}
 	}
 
-	// Remove orphaned routes from host network.
-	actualRouteMap, err := c.listIPRoutes()
+	// Remove any unknown routes on antrea-gw0.
+	routes, err := c.listIPRoutesOnGW()
 	if err != nil {
 		return fmt.Errorf("error listing ip routes: %v", err)
 	}
-	for podCIDR, actualRoutes := range actualRouteMap {
-		if desiredPodCIDRs.Has(podCIDR) {
+	for _, route := range routes {
+		if reflect.DeepEqual(route.Dst, c.nodeConfig.PodCIDR) {
 			continue
 		}
-		for _, route := range actualRoutes {
-			klog.V(4).Infof("Deleting orphaned route %v", route)
-			if err := netlink.RouteDel(route); err != nil && err != unix.ESRCH {
-				return err
-			}
+		if desiredPodCIDRs.Has(route.Dst.String()) {
+			continue
+		}
+		klog.Infof("Deleting unknown route %v", route)
+		if err := netlink.RouteDel(&route); err != nil && err != unix.ESRCH {
+			return err
 		}
 	}
 	return nil
 }
 
+// listIPRoutes returns list of routes on antrea-gw0.
+func (c *Client) listIPRoutesOnGW() ([]netlink.Route, error) {
+	// get all routes on antrea-gw0 from service table.
```

Comment:
Nit: get -> Get

Reply:
I thought we no more use "service table"?

Reply:
removed.